designer/public/js/anomaly.js

```diff
@@ -105,25 +112,47 @@ $(function() {
             console.log('failed to query context');
         });
     }
-
     function checkIntent() {
         var queryReq = {};
-        queryReq.entities = [{ type: 'ServiceIntent', isPattern: true }];
-        queryReq.restriction = { scopes: [{ scopeType: 'stringQuery', scopeValue: 'topology=Topology.anomaly-detection' }] }
-
-        client.queryContext(queryReq).then(function(resultList) {
+       // queryReq.entities = [{ type: 'ServiceIntent', isPattern: true }];
+        //queryReq.restriction = { scopes: [{ scopeType: 'stringQuery', scopeValue: 'topology=Topology.anomaly-detection' }] }
+        queryReq = { internalType: "ServiceIntent", updateAction: "UPDATE" };
+        scopeValue = 'anomaly-detection'
+        clientDes.getContext(queryReq).then(function(resultList) {
             console.log(resultList);
-            if (resultList && resultList.length > 0) {
-                curIntent = resultList[0];
-
-                //update the current geoscope as well
-                geoscope = curIntent.attributes.intent.geoscope;
+            if (resultList.data && resultList.data.length > 0) {
+                var result = resultList.data.filter(x => x.topology === scopeValue);
+                if (result.length > 0) {
+                    console.log("service intent result ---- ",result);
+                    curIntent = result[0];
+                    //update the current geoscope as well
+                    geoscope = result[0].geoscope;
+                }
+                
             }
         }).catch(function(error) {
             console.log(error);
             console.log('failed to query context');
         });
     }
+    // function checkIntent() {
```

Comment:
better to write a comment on the top of this commented section why this is commented.